biz.aQute.bndlib/src/aQute/bnd/osgi/repository/AggregateRepository.java

```diff
@@ -0,0 +1,53 @@
+package aQute.bnd.osgi.repository;
+
+import static java.util.Collections.singleton;
+
+import java.util.Collection;
+import java.util.Collections;
+import java.util.Map;
+
+import org.osgi.resource.Capability;
+import org.osgi.resource.Requirement;
+import org.osgi.service.repository.Repository;
+
+import aQute.lib.collections.MultiMap;
+
+public class AggregateRepository implements Repository {
+
+	private final Repository repositories[];
+
+	public AggregateRepository(Collection< ? extends Repository> repositories) {
+		this(repositories.toArray(new Repository[repositories.size()]));
+	}
+
+	public AggregateRepository(Repository... repositories) {
+		this.repositories = repositories;
```

Comment:
Hmm, normally I do not care that much but I actually got bitten this morning by a list that I did not copy here and that was later cleared. That was the reason for the array. But I've fixed it.